bundles/auth/core/src/main/java/org/apache/sling/auth/core/impl/SlingAuthenticator.java

```diff
@@ -1657,11 +1657,15 @@ protected AbstractAuthenticationHandlerHolder createHolder(String path,
         }
 
         boolean isNodeRequiresAuthHandler() {
-            if (requestPath.equalsIgnoreCase(handlerPath)) {
+            final String requestPathNoSelectors = requestPath.contains(".") ?
```

Comment:
makes sense. An alternative would be to split the request path into an array by each level in the repository (split by "/"). Also split the handler path in a similar way and see if the last 2 nodes on each are on the same level and not equal.